python/mypy_protobuf.py

```diff
@@ -528,6 +529,77 @@ def write_services(self, services):
                 )
                 self.write_methods(service, is_abstract=False)
 
+    def _output_type(self, method):
+        # type: (d.MethodDescriptorProto) -> Text
+        result = self._import_message(method.output_type)
+        if method.server_streaming:
+            result = "{}[{}, None, None]".format(
+                self._import("typing", "Generator"), result,
+            )
+        return result
+
+    def write_grpc_methods(self, service):
+        # type: (d.ServiceDescriptorProto) -> None
+        l = self._write_line
+        methods = [m for m in service.method if m.name not in PYTHON_RESERVED]
+        if not methods:
+            l("pass")
+            l("")
+        for method in methods:
+            l("@{}", self._import("abc", "abstractmethod"))
+            l("def {}(self,", method.name)
+            with self._indent():
+                l("request: {},", self._import_message(method.input_type))
+                l("context: {},", self._import("grpc", "ServicerContext"))
+            l(
+                ") -> {}: ...", self._output_type(method),
+            )
+            l("")
+
+    def write_grpc_stub_methods(self, service):
+        # type: (d.ServiceDescriptorProto) -> None
+        l = self._write_line
+        methods = [m for m in service.method if m.name not in PYTHON_RESERVED]
+        if not methods:
+            l("pass")
+            l("")
+        for method in methods:
+            l("def {}(self,", method.name)
+            with self._indent():
+                l("request: {},", self._import_message(method.input_type))
+            l(
+                ") -> {}: ...", self._output_type(method),
+            )
+            l("")
+
+    def write_grpc_services(self, services):
+        # type: (Iterable[d.ServiceDescriptorProto]) -> None
+        l = self._write_line
+        l(
+            "from .{} import *",
+            self.fd.name.rsplit('/', 1)[1][:-6].replace("-", "_") + "_pb2",
```

Comment:
Will probably not work if there is no / inside the file name, i.e. dealing with top-level proto files. Why not using something like os.path.basename?

Reply:
Actually seems like using [-1] will solve this issue

Reply:
I actually used `Path(self.fd.name).stem`.
this removes the whole cryptic indices.

Reply:
@Zahlii Thanks. I will fix.